### PR TITLE
Added two special cases to the Model Registry Workflow API

### PR DESCRIPTION
**Pull request by dmatrix**
## What changes are proposed in this pull request?

This PR adds two scenarios to the Model Registry workflow API:
 

- How to register existing saved models in a native format not saved or trained with MLflow
- How to register models with no MLflow built-in model flavor support.

- [ x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [x ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: docs/source/model-registry.rst [hunk 1]
```diff
@@ -349,3 +349,256 @@ You can either delete specific versions of a registered model or you can delete
...
+
+.. code-block:: text
+
+    --
+    Successfully registered model 'SklearnLinearRegression'.
+    2021/04/02 16:30:57 INFO mlflow.tracking._model_registry.client: Waiting up to 300 seconds for model version to finish creation.
+    Model name: SklearnLinearRegression, version 1
+    Created version '1' of model 'SklearnLinearRegression'.
+
+Now, using MLflow fluent APIs, we reload the from the Model Registry and score.
```
Comment (sueann) [inline docs/source/model-registry.rst:456]: do we need this section specifically here? seems generic for all registered models. or is it to specifically show that the registered model can be applied to the exact same data set as the originally saved model?

Comment (sueann) [reply docs/source/model-registry.rst:456]: i guess it does not hurt to have this so there is an end to end workflow example here, just was worried about the length

**Review by sueann — COMMENTED** [review]
the content generally looks good to me - thanks for doing this @dmatrix ! cc @andreakress for the language and possibly organization.

File: docs/source/model-registry.rst (hunk 1 above)
Comment (dmatrix) [reply docs/source/model-registry.rst:456]: Yes, @sueann, I put it in there for that purpose. In this case, I was comparing and verifying that the model saved without using MLflow and then logged and registered with MLflow produces the same predictions.

**Comment by andreakress** [comment]
LGTM, very clearly written. Couple of style suggestions, I might not start a sentence with "And..." and I might not use "wish" but they don't detract from the content at all so up to you.

**Comment by dmatrix** [comment]
Thankyou @andreakress. Noted your style suggestions and incorporated.

**Review by sueann — APPROVED** [review]

**Pull request merged**